x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/PotentiallyUnmappedKeywordEsField.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.esql.core.type;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+
+/** Information about a field in an ES index that may not mapped. If it is unmapped, it should be loaded from source directly. */
+public class PotentiallyUnmappedKeywordEsField extends KeywordEsField {
```

Comment:
The field will be unmapped in at least one index hence its potential has been realized :) and the class can be simply called UnmappedKeywordEsField